test/v1beta1/main_test.go

```diff
@@ -0,0 +1,180 @@
+package v1beta1_test
+
+import (
+	"context"
+	"fmt"
+	fake "github.com/brianvoe/gofakeit/v5"
+	"github.com/grafeas/grafeas/proto/v1beta1/build_go_proto"
+	"github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
+	"github.com/grafeas/grafeas/proto/v1beta1/project_go_proto"
+	"github.com/grafeas/grafeas/proto/v1beta1/provenance_go_proto"
+	. "github.com/onsi/gomega"
+	"google.golang.org/grpc"
+	"log"
+	"testing"
+)
+
+type setup struct {
+	ctx context.Context
+	gc  grafeas_go_proto.GrafeasV1Beta1Client
+	pc  project_go_proto.ProjectsClient
+}
+
+func newSetup() (*setup, error) {
+	ctx := context.Background()
+
+	projectsCC, err := grpc.DialContext(ctx, "localhost:8080", grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	projectsClient := project_go_proto.NewProjectsClient(projectsCC)
+
+	grafeasCC, err := grpc.DialContext(ctx, "localhost:8080", grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	grafeasClient := grafeas_go_proto.NewGrafeasV1Beta1Client(grafeasCC)
+
+	return &setup{
+		ctx,
+		grafeasClient,
+		projectsClient,
+	}, nil
+}
+
+func TestGrafeasElasticsearch(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("Test run with -short flag, skipping.")
+	}
+
+	fake.Seed(0)
+
+	g := NewGomegaWithT(t)
+	Expect := g.Expect
+
+	s, err := newSetup()
+	if err != nil {
+		log.Fatalf("Failed to create test setup.\nError: %v", err)
+	}
+
+	t.Run("projects", func(t *testing.T) {
+		t.Run("creating a project", func(t *testing.T) {
+			t.Run("should succeed with a valid name", func(t *testing.T) {
+				name := randomProjectName()
+
+				p, err := s.pc.CreateProject(s.ctx, &project_go_proto.CreateProjectRequest{Project: &project_go_proto.Project{Name: name}})
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = s.pc.GetProject(s.ctx, &project_go_proto.GetProjectRequest{Name: p.GetName()})
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		t.Run("listing projects", func(t *testing.T) {
+			type test struct {
+				name, filter string
+				expected     []string
+			}
+
+			names := []string{
+				"projects/foo",
+				"projects/bar",
+				"projects/foo-bar-123",
+			}
+
+			for _, n := range names {
+				_, err = s.pc.CreateProject(s.ctx, &project_go_proto.CreateProjectRequest{Project: &project_go_proto.Project{Name: n}})
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			t.Run("filters", func(t *testing.T) {
+				for _, tc := range []test{
+					{
+						name:     "single exact name match",
+						filter:   "name==\"projects/foo\"",
+						expected: []string{"projects/foo"},
+					},
+					{
+						name:     "or exact name match",
+						filter:   "name==\"projects/foo\"||name==\"projects/bar\"",
+						expected: []string{"projects/foo", "projects/bar"},
+					},
+				} {
+					tc := tc
```

Comment:
why is this needed?

Reply:
The tests in this table are run in parallel. If I don't shadow `tc`, or wrap each test in its own `func` call, they will not execute with the correct testcase.